src/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -23,30 +23,60 @@ namespace System
 {
     public static class Buffer
     {
+        /// <summary>
+        /// This field is used to quickly check whether an array that is given to the BlockCopy
+        /// method is a byte array and the code can take optimized path.
+        /// </summary>
+        private static EETypePtr s_byteArrayEEType;
+
+        internal static void Initialize()
```

Comment:
This is leaf that is not order dependent, so it can be just marked with EagerStaticClassConstructionAttribute.